cygnss_wetlands/animation/animate.py

```diff
@@ -0,0 +1,346 @@
+        elif latitude < 0:
+            yticks.append(f"{abs(latitude)}°S")
+        else:
+            yticks.append(f"{abs(latitude)}°")
+
+    xticks = []
+    for longitude in longitudes:
+        if latitude > 0:
+            xticks.append(f"{abs(longitude)}°E")
+        elif latitude < 0:
+            xticks.append(f"{abs(longitude)}°W")
+        else:
+            xticks.append(f"{abs(longitude)}°")
+
+    # Set the ticks on the X & Y axis to be the representative long/lat coordinates
+    ax.set_xticks(x_ticks, xticks)
+    ax.set_yticks(y_ticks, yticks)
+
+    plt.savefig(figureName)
+
+
+def animate(
+    startDate: datetime,
+    endDate: datetime,
+    plotVariable: str,
+    outputFileLocation: Path = Path("./"),
+    monthlyIntervals: int = 2,
+    frameDuration: int = 1000,
+    gridType: GridType = GridType.EASE2_G9km,
+    generateDistribution: bool = False,
+):
+    """
+    Creates and saves a GIF animation by creating monthlyIntervals of figures per month from the startDate to endDate.
+
+    Args:
+        startDate (datetime): Start date of animation. Animation will begin at the beginning of the selected month.
+        endDate (datetime): End date of animation. Animation will end at the end of the selected month.
+        plotVariable (str): Variable to plot
+        outputFileLocation (Path): Root directory for figure & animation directories (default=Path("./"))
+        monthlyIntervals (int): Number of frames per month of animation (default=2)
+        frameDuration (int): ms between frames (default=1000)
+        gridType (GridType): EASE2 GridType describing size of pixel footprint (default=EASE2_G9km)
+        generateDistribution (bool): Boolean to generate a histogram of the dataset (default=False)
+
+    Returns:
+        None: Saves GIF animation to outputFileLocation/animations/VARIABLENAME_YYYYMMDD(start)-YYYMMDD(end)_gridResolution_monthlyIntervals.gif
+    """
+
+    figures = []
+    grid = EASE2GRID(gridType)
+
+    figurePath, animationPath = generateAnimationFileStructure(destination_dir=outputFileLocation)
+
+    file_base = (
+        plotVariable.upper()
+        + "_"
+        + str(startDate.year)
+        + f"{startDate.month:02}"
+        + "-"
+        + str(endDate.year)
+        + f"{endDate.month:02}"
+        + "_"
+        + str(int(grid.res))
+        + "_"
+        + str(monthlyIntervals)
+    )
+
+    gif_name = file_base + ".gif"
+    gif_path = Path.joinpath(animationPath, gif_name)
+
+    if not gif_path.exists():
+        print(f"Generating Animation {gif_name}")
+
+        # Collect metadata on the requested variable
+        print(f"Calculating metadata for {plotVariable}")
+        metadata = reader.metadata(variable_name=plotVariable, grid=grid, start_date=startDate, end_date=endDate)
+        min = 0
+        # Select only the top 99% of data to extract outliers; used for the colorbar max
+        max = round(np.nanpercentile(metadata["data"], 99))
```

Comment:
Maybe 99th percentile should be calculated in the metadata function? and so that you don't need to store a copy of the entire variable array here